### ISSUE TITLE
Faster RCNN training exited with Process received signal

### ISSUE THREAD

**Issue by xinli94** [tensorpack/tensorpack#1031]
If you're asking about an unexpected problem you met, use this template.
__PLEASE DO NOT DELETE THIS TEMPLATE, FILL IT__:

### 1. What you did:

(1) **If you're using examples, what's the command you run:**
Faster RCNN
`CUDA_VISIBLE_DEVICES=5,6,7,8 NCCL_DEBUG=INFO mpirun -np 4 ./train.py --load pretrained/COCO-R101FPN-MaskRCNN-Standard_filtered.npz --logdir /data5/xin/logo_jv_48/resnet101_fpn --config MODE_MASK=False MODE_FPN=True TRAINER=horovod |& grep -v "Read -1"`

(2) **If you're using examples, have you made any changes to the examples? Paste them here:**
- Modified `config.py` to use a custom dataset
- Removed `fastrcnn/outputs/,maskrcnn/conv/` from official pretrained ckpts, and used that to do transfer learning. (I refered to `https://github.com/tensorpack/tensorpack/issues/1015`)
- Also I am loading a mask+fpn ckpt into a fpn only model

(3) **If not using examples, tell us what you did here:**
N/A

### 2. What you observed:

(1) **Include the ENTIRE logs here:**

It's always better to paste what you observed instead of describing them.

A part of logs is sometimes enough, but it's always better to paste as much as possible.

You can run a command with `CMD 2>&1 | tee logs.txt` to save all stdout & stderr logs to one file.
```
[0104 13:46:38 @base.py:284] Epoch 160 (global_step 80000) finished, time:2 minutes 55 seconds.
[0104 13:46:38 @trainers.py:417] Running horovod broadcast ...
[0104 13:46:39 @train.py:490] Running evaluation ...
[0104 13:46:39 @train.py:490] Running evaluation ...
[0104 13:46:39 @train.py:490] Running evaluation ...

  0%|          |0/12[00:00<?,?it/s]
  0%|          |0/12[00:00<?,?it/s]
  0%|          |0/12[00:00<?,?it/s][0104 13:46:40 @saver.py:79] Model saved to /data5/xin/logo_jv_48/resnet101_fpn/model-80000.
[0104 13:46:40 @train.py:490] Running evaluation ...

100%|##########|12/12[00:01<00:00, 8.24it/s][0104 13:46:41 @monitor.py:467] PeakMemory(MB)/gpu:0: 7719.4
[0104 13:46:41 @group.py:48] Callbacks took 3.134 sec in total. EvalCallback: 2.23 seconds
[0104 13:46:41 @base.py:274] Start Epoch 161 ...
[0104 13:46:41 @monitor.py:467] PeakMemory(MB)/gpu:0: 7720.1
[0104 13:46:41 @group.py:48] Callbacks took 3.133 sec in total. EvalCallback: 2.23 seconds
[0104 13:46:41 @monitor.py:467] PeakMemory(MB)/gpu:0: 7770.4
[0104 13:46:41 @base.py:274] Start Epoch 161 ...
[0104 13:46:41 @group.py:48] Callbacks took 3.133 sec in total. EvalCallback: 2.23 seconds
[0104 13:46:41 @base.py:274] Start Epoch 161 ...

  0%|          |0/500[00:00<?,?it/s]
  0%|          |0/500[00:00<?,?it/s]
  0%|          |0/500[00:00<?,?it/s]loading annotations into memory...
Done (t=0.00s)
creating index...
index created!
Loading and preparing results...
DONE (t=0.00s)
creating index...
index created!
Running per image evaluation...
Evaluate annotation type *bbox*
DONE (t=0.06s).
Accumulating evaluation results...
DONE (t=0.19s).
 Average Precision  (AP) @[ IoU=0.50:0.95 | area=   all | maxDets=100 ] = 0.665
 Average Precision  (AP) @[ IoU=0.50      | area=   all | maxDets=100 ] = 0.891
 Average Precision  (AP) @[ IoU=0.75      | area=   all | maxDets=100 ] = 0.714
 Average Precision  (AP) @[ IoU=0.50:0.95 | area= small | maxDets=100 ] = 0.600
 Average Precision  (AP) @[ IoU=0.50:0.95 | area=medium | maxDets=100 ] = 0.652
 Average Precision  (AP) @[ IoU=0.50:0.95 | area= large | maxDets=100 ] = 0.698
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets=  1 ] = 0.654
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets= 10 ] = 0.690
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets=100 ] = 0.690
 Average Recall     (AR) @[ IoU=0.50:0.95 | area= small | maxDets=100 ] = 0.600
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=medium | maxDets=100 ] = 0.690
 Average Recall     (AR) @[ IoU=0.50:0.95 | area= large | maxDets=100 ] = 0.704
[0104 13:46:41 @misc.py:109] Estimated Time Left: 2 days 18 hours 14 minutes 24 seconds
[0104 13:46:41 @monitor.py:467] PeakMemory(MB)/gpu:0: 7839.2
[0104 13:46:41 @monitor.py:467] QueueInput/queue_size: 50
[0104 13:46:41 @monitor.py:467] fastrcnn_losses/box_loss: 0.021068
[0104 13:46:41 @monitor.py:467] fastrcnn_losses/label_loss: 0.046407
[0104 13:46:41 @monitor.py:467] fastrcnn_losses/label_metrics/accuracy: 0.98228
[0104 13:46:41 @monitor.py:467] fastrcnn_losses/label_metrics/false_negative: 0.10834
[0104 13:46:41 @monitor.py:467] fastrcnn_losses/label_metrics/fg_accuracy: 0.89153
[0104 13:46:41 @monitor.py:467] fastrcnn_losses/num_fg_label: 30.403
[0104 13:46:41 @monitor.py:467] learning_rate: 0.01
[0104 13:46:41 @monitor.py:467] mAP(bbox)/IoU=0.5: 0.89062
[0104 13:46:41 @monitor.py:467] mAP(bbox)/IoU=0.5:0.95: 0.6651
[0104 13:46:41 @monitor.py:467] mAP(bbox)/IoU=0.75: 0.71354
[0104 13:46:41 @monitor.py:467] mAP(bbox)/large: 0.69792
[0104 13:46:41 @monitor.py:467] mAP(bbox)/medium: 0.6525
[0104 13:46:41 @monitor.py:467] mAP(bbox)/small: 0.6
[0104 13:46:41 @monitor.py:467] multilevel_roi_align/fpn_map_rois_to_levels/num_roi_level2: 356.93
[0104 13:46:41 @monitor.py:467] multilevel_roi_align/fpn_map_rois_to_levels/num_roi_level3: 93.572
[0104 13:46:41 @monitor.py:467] multilevel_roi_align/fpn_map_rois_to_levels/num_roi_level4: 53.026
[0104 13:46:41 @monitor.py:467] multilevel_roi_align/fpn_map_rois_to_levels/num_roi_level5: 8.4725
[0104 13:46:41 @monitor.py:467] rpn_losses/box_loss: 0.0039561
[0104 13:46:41 @monitor.py:467] rpn_losses/label_loss: 0.0040023
[0104 13:46:41 @monitor.py:467] rpn_losses/level2/box_loss: 0.00093638
[0104 13:46:41 @monitor.py:467] rpn_losses/level2/label_loss: 0.002108
[0104 13:46:41 @monitor.py:467] rpn_losses/level2/label_metrics/precision_th0.1: 0.50178
[0104 13:46:41 @monitor.py:467] rpn_losses/level2/label_metrics/precision_th0.2: 0.5272
[0104 13:46:41 @monitor.py:467] rpn_losses/level2/label_metrics/precision_th0.5: 0.54886
[0104 13:46:41 @monitor.py:467] rpn_losses/level2/label_metrics/recall_th0.1: 0.56901
[0104 13:46:41 @monitor.py:467] rpn_losses/level2/label_metrics/recall_th0.2: 0.5653
[0104 13:46:41 @monitor.py:467] rpn_losses/level2/label_metrics/recall_th0.5: 0.55671
[0104 13:46:41 @monitor.py:467] rpn_losses/level2/num_pos_anchor: 1.7738
[0104 13:46:41 @monitor.py:467] rpn_losses/level2/num_valid_anchor: 192.01
[0104 13:46:41 @monitor.py:467] rpn_losses/level3/box_loss: 0.0011188
[0104 13:46:41 @monitor.py:467] rpn_losses/level3/label_loss: 0.00067379
[0104 13:46:41 @monitor.py:467] rpn_losses/level3/label_metrics/precision_th0.1: 0.6187
[0104 13:46:41 @monitor.py:467] rpn_losses/level3/label_metrics/precision_th0.2: 0.65407
[0104 13:46:41 @monitor.py:467] rpn_losses/level3/label_metrics/precision_th0.5: 0.66024
[0104 13:46:41 @monitor.py:467] rpn_losses/level3/label_metrics/recall_th0.1: 0.66768
[0104 13:46:41 @monitor.py:467] rpn_losses/level3/label_metrics/recall_th0.2: 0.66768
[0104 13:46:41 @monitor.py:467] rpn_losses/level3/label_metrics/recall_th0.5: 0.66627
[0104 13:46:41 @monitor.py:467] rpn_losses/level3/num_pos_anchor: 3.6392
[0104 13:46:41 @monitor.py:467] rpn_losses/level3/num_valid_anchor: 48.663
[0104 13:46:41 @monitor.py:467] rpn_losses/level4/box_loss: 0.00080095
[0104 13:46:41 @monitor.py:467] rpn_losses/level4/label_loss: 0.00095062
[0104 13:46:41 @monitor.py:467] rpn_losses/level4/label_metrics/precision_th0.1: 0.6574
[0104 13:46:41 @monitor.py:467] rpn_losses/level4/label_metrics/precision_th0.2: 0.65741
[0104 13:46:41 @monitor.py:467] rpn_losses/level4/label_metrics/precision_th0.5: 0.66659
[0104 13:46:41 @monitor.py:467] rpn_losses/level4/label_metrics/recall_th0.1: 0.68995
[0104 13:46:41 @monitor.py:467] rpn_losses/level4/label_metrics/recall_th0.2: 0.68995
[0104 13:46:41 @monitor.py:467] rpn_losses/level4/label_metrics/recall_th0.5: 0.68385
[0104 13:46:41 @monitor.py:467] rpn_losses/level4/num_pos_anchor: 2.3473
[0104 13:46:41 @monitor.py:467] rpn_losses/level4/num_valid_anchor: 11.72
[0104 13:46:41 @monitor.py:467] rpn_losses/level5/box_loss: 0.00074321
[0104 13:46:41 @monitor.py:467] rpn_losses/level5/label_loss: 0.00015627
[0104 13:46:41 @monitor.py:467] rpn_losses/level5/label_metrics/precision_th0.1: 0.63178
[0104 13:46:41 @monitor.py:467] rpn_losses/level5/label_metrics/precision_th0.2: 0.63226
[0104 13:46:41 @monitor.py:467] rpn_losses/level5/label_metrics/precision_th0.5: 0.63623
[0104 13:46:41 @monitor.py:467] rpn_losses/level5/label_metrics/recall_th0.1: 0.63804
[0104 13:46:41 @monitor.py:467] rpn_losses/level5/label_metrics/recall_th0.2: 0.63804
[0104 13:46:41 @monitor.py:467] rpn_losses/level5/label_metrics/recall_th0.5: 0.63804
[0104 13:46:41 @monitor.py:467] rpn_losses/level5/num_pos_anchor: 1.5639
[0104 13:46:41 @monitor.py:467] rpn_losses/level4/label_metrics/precision_th0.2: 0.65741
[0104 13:46:41 @monitor.py:467] rpn_losses/level4/label_metrics/precision_th0.5: 0.66659
[0104 13:46:41 @monitor.py:467] rpn_losses/level4/label_metrics/recall_th0.1: 0.68995
[0104 13:46:41 @monitor.py:467] rpn_losses/level4/label_metrics/recall_th0.2: 0.68995
[0104 13:46:41 @monitor.py:467] rpn_losses/level4/label_metrics/recall_th0.5: 0.68385
[0104 13:46:41 @monitor.py:467] rpn_losses/level4/num_pos_anchor: 2.3473
[0104 13:46:41 @monitor.py:467] rpn_losses/level4/num_valid_anchor: 11.72
[0104 13:46:41 @monitor.py:467] rpn_losses/level5/box_loss: 0.00074321
[0104 13:46:41 @monitor.py:467] rpn_losses/level5/label_loss: 0.00015627
[0104 13:46:41 @monitor.py:467] rpn_losses/level5/label_metrics/precision_th0.1: 0.63178
[0104 13:46:41 @monitor.py:467] rpn_losses/level5/label_metrics/precision_th0.2: 0.63226
[0104 13:46:41 @monitor.py:467] rpn_losses/level5/label_metrics/precision_th0.5: 0.63623
[0104 13:46:41 @monitor.py:467] rpn_losses/level5/label_metrics/recall_th0.1: 0.63804
[0104 13:46:41 @monitor.py:467] rpn_losses/level5/label_metrics/recall_th0.2: 0.63804
[0104 13:46:41 @monitor.py:467] rpn_losses/level5/label_metrics/recall_th0.5: 0.63804
[0104 13:46:41 @monitor.py:467] rpn_losses/level5/num_pos_anchor: 1.5639
[0104 13:46:41 @monitor.py:467] rpn_losses/level5/num_valid_anchor: 2.9051
[0104 13:46:41 @monitor.py:467] rpn_losses/level6/box_loss: 0.00035671
[0104 13:46:41 @monitor.py:467] rpn_losses/level6/label_loss: 0.00011368
[0104 13:46:41 @monitor.py:467] rpn_losses/level6/label_metrics/precision_th0.1: 0.55148
[0104 13:46:41 @monitor.py:467] rpn_losses/level6/label_metrics/precision_th0.2: 0.55148
[0104 13:46:41 @monitor.py:467] rpn_losses/level6/label_metrics/precision_th0.5: 0.55341
[0104 13:46:41 @monitor.py:467] rpn_losses/level6/label_metrics/recall_th0.1: 0.57179
[0104 13:46:41 @monitor.py:467] rpn_losses/level6/label_metrics/recall_th0.2: 0.57179
[0104 13:46:41 @monitor.py:467] rpn_losses/level6/label_metrics/recall_th0.5: 0.57179
[0104 13:46:41 @monitor.py:467] rpn_losses/level6/num_pos_anchor: 0.52498
[0104 13:46:41 @monitor.py:467] rpn_losses/level6/num_valid_anchor: 0.70655
[0104 13:46:41 @monitor.py:467] sample_fast_rcnn_targets/num_bg: 481.6
[0104 13:46:41 @monitor.py:467] sample_fast_rcnn_targets/num_fg: 30.403
[0104 13:46:41 @monitor.py:467] sample_fast_rcnn_targets/proposal_metrics/best_iou_per_gt: 0.85207
[0104 13:46:41 @monitor.py:467] sample_fast_rcnn_targets/proposal_metrics/recall_iou0.3: 0.99994
[0104 13:46:41 @monitor.py:467] sample_fast_rcnn_targets/proposal_metrics/recall_iou0.5: 0.99994
[0104 13:46:41 @monitor.py:467] total_cost: 0.18447
[0104 13:46:41 @monitor.py:467] wd_cost: 0.10904
[0104 13:46:41 @group.py:48] Callbacks took 3.443 sec in total. EvalCallback: 1.72 seconds
[0104 13:46:41 @base.py:274] Start Epoch 161 ...

[65b89549d1f6:57689] *** Process received signal ***
```

(2) **Other observations, if any:**
For example, CPU/GPU utilization, output images, tensorboard curves, if relevant to your issue.
N/A

### 3. What you expected, if not obvious.
- The training crashed after 160 epochs without any useful error message. According to the math here `https://github.com/tensorpack/tensorpack/blob/master/examples/FasterRCNN/train.py#L597`, `max_epoch=360000 * (8/4) * 500 = 1440`. I am not sure why it stopped at epoch 160.
- I restarted the training from epoch 1 and it crashed after 80 epochs with `Process received signal` again.
- I am wondering if this problem is tensorpack-related or not, and if there is any suggestion on debugging this!

### 4. Your environment:
  + Python version: - `3.5.2`
  + TF version: `python -c 'import tensorflow as tf; print(tf.GIT_VERSION, tf.VERSION)'`. - `v1.12.0-0-ga6d8ffae09 1.12.0`
  + Tensorpack version: `python -c 'import tensorpack; print(tensorpack.__version__);'`.
      You can install Tensorpack master by `pip install -U git+https://github.com/ppwwyyxx/tensorpack.git`
      and see if your issue is already solved. - `0.9.0.1`
  + If you're not using tensorpack under a normal command line shell (e.g.,
    using an IDE or jupyter notebook), please retry under a normal command line shell. 
  + Hardware information, e.g. number of GPUs used. -  `4 TITAN Xp`

**Comment by ppwwyyxx** [comment]
I very occasionally saw strange crashes when running with horovod, but never when using the "replicated" trainer. My guess is this is horovod-specific, but I cannot tell because I can never reproduce such errors. https://github.com/uber/horovod/issues/404 is one such example and I've seen other crashes with different behaviors. Unless you want to do distributed training, using the "replicated" trainer is the same as horovod.

You can also check more logs: in addition to the logs saved by tensorpack, `mpirun` also saves logs for you and may contain more information.
Also, your job can crash if CPU memory is not enough.

**Comment by xinli94** [comment]
I trained `res101_cascade_fpn_gn` from scratch and `replicated` trainer works perfectly for me! Closing the issue now. Will post here if I find anything.
